### PR TITLE
virttest.ip_sniffing: Wait for "self._process" to be set

### PR DESCRIPTION
**Pull request by ldoktor**
It might happen that the reader thread is faster than the main thread
which sets the "self._process", which then crashes as "self._process" is
not yet set. Let's allow waiting for it for a while.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

**Comment by ldoktor** [comment]
A reproducer is to use "user" (not root) and "tcpdump" which requires root.

File: virttest/ip_sniffing.py [hunk 1]
```diff
@@ -167,7 +167,10 @@ def _output_logger_handler(self, line):
         # We can check whether the process is terminated unexpectedly
         # here since the terminated status is a line of the output
         if re.match("\(Process terminated with status", line):
-            status = self._process.get_status()
+            if wait_for(lambda: self._process, 1, 0.1):
+                status = self._process.get_status()
+            else:
+                status = "<not available, self._process not set>"
```
Comment (luckyh) [inline virttest/ip_sniffing.py]: How about capturing the `status` from log message, like:
```python
match = re.match("\(Process terminated with status (\d+)", line):
if match:
    status = match.group(1)
```
thus it won't reference `self._process` any more.

Comment (ldoktor) [reply virttest/ip_sniffing.py]: yep, makes sense and the probability of false-match is really low (that's what would be indicated by "not available..."). I'll change it (probably tomorrow, right now I have some other business)

**Comment by ldoktor** [comment]
OK, updated and added new commit to compile the regexp.

**Pull request merged**